internal/command/doctor/doctor.go

```diff
@@ -253,3 +358,86 @@ func runLocalDocker(ctx context.Context) (err error) {
 
 	return
 }
+
+func getFirstFlyDevNameserver(dnsClient *dns.Client) (string, error) {
+	msg := &dns.Msg{}
+	flydev := "fly.dev"
+	msg.SetQuestion(dns.Fqdn(flydev), dns.TypeNS)
+	msg.RecursionDesired = true
+	// TODO: use ipv6 when system supports it
+	resolver := "9.9.9.9:53"
+	r, _, err := dnsClient.Exchange(msg, resolver)
+	if err != nil {
+		return "", err
+	}
+	if r.Rcode != dns.RcodeSuccess {
+		return "", fmt.Errorf("failed to resolve NS record for %s. Got error code: %s", flydev, dns.RcodeToString[r.Rcode])
+	}
+	for _, a := range r.Answer {
+		if ns, ok := a.(*dns.NS); ok {
+			return ns.Ns, nil
+		}
+	}
+	return "", fmt.Errorf("no NS records found for %s", flydev)
+}
+
+func checkDnsRecords(dnsClient *dns.Client, nsAddr string, appName string, appFqdn string, qType string, appIps map[string]bool) (error, string) {
+	msg := &dns.Msg{}
+	msg.SetQuestion(appFqdn, dns.StringToType[qType])
+	msg.RecursionDesired = true
+
+	r, _, err := dnsClient.Exchange(msg, nsAddr)
+	if err != nil {
+		return fmt.Errorf("failed to lookup A record for %s: %w", appFqdn, err), ""
+	}
+	if r.Rcode != dns.RcodeSuccess {
+		return fmt.Errorf("invalid result when looking up A record for %s: %s", appFqdn, dns.RcodeToString[r.Rcode]), ""
+	}
+	dnsIps := make(map[string]bool)
+	for _, a := range r.Answer {
+		if qType == "A" {
+			if aRec, ok := a.(*dns.A); ok {
+				dnsIps[aRec.A.String()] = true
+			}
+		} else if qType == "AAAA" {
+			if aRec, ok := a.(*dns.AAAA); ok {
+				dnsIps[aRec.AAAA.String()] = true
+			}
+		}
+	}
+
+	ipsOnAppNotInDns := make([]string, 0)
+	for appIp := range appIps {
+		if _, present := dnsIps[appIp]; !present {
+			ipsOnAppNotInDns = append(ipsOnAppNotInDns, appIp)
+		}
+	}
+	ipsInDnsNotInApp := make([]string, 0)
+	for dnsIp := range dnsIps {
+		if _, present := appIps[dnsIp]; !present {
+			ipsInDnsNotInApp = append(ipsInDnsNotInApp, dnsIp)
+		}
+	}
+
+	if len(ipsOnAppNotInDns) == 0 && len(ipsInDnsNotInApp) == 0 {
+		return nil, ""
+	} else if len(ipsOnAppNotInDns) > 0 {
+		missingIps := strings.Join(ipsOnAppNotInDns, ", ")
+		return fmt.Errorf(`Nope
+	These IPs are missing from the %s %s record: %s
+	This likely means we had an operational issue when we tried to create the record.
+	You can force an update by running:
```

Comment:
arguably we don't want this shell monstrosity in our actual output, and we can just put a link to a forum thingy here